sonar-scm-git-plugin/src/main/java/org/sonarsource/scm/git/GitScmProvider.java

```diff
@@ -91,6 +91,22 @@ public BlameCommand blameCommand() {
     return null;
   }
 
+  @Override
+  public Path relativePathFromScmRoot(Path path) {
+    RepositoryBuilder builder = getVerifiedRepositoryBuilder(path);
+    return builder.getGitDir().toPath().getParent().relativize(path);
+  }
+
+  @Override
+  public String revisionId(Path path) {
+    RepositoryBuilder builder = getVerifiedRepositoryBuilder(path);
+    try {
+      return builder.build().exactRef("HEAD").getObjectId().getName();
+    } catch (IOException e) {
+      throw MessageException.of("I/O error while getting revision ID for path: " + path, e);
```

Comment:
The error looks really technical. Why using a MessageException?

Reply:
I sort of blindly followed the example of another exception in the same file. I also don't know what would be a more appropriate exception here. Do you have a better idea?

Reply:
I usually convert IOException to IllegalStateException, but this is just a personal habit.